### PR TITLE
Add a procfs payload

### PR DESCRIPTION
**Pull request by goxberry**
### What does this PR do?

This pull request adds a collection of types that model the following `/proc` filesystem files for a given process ID `[pid]`

* `/proc/[pid]/cmdline`
* `/proc/[pid]/comm`
* `/proc/[pid]/io`
* `/proc/[pid]/stat`
* `/proc/[pid]/statm`
* `/proc/[pid]/status`

along with methods to generate random approximately valid data for each of these files. These files are used by `process-agent` to observe process resource consumption.

In addition, this pull request begins implementing a `procfs` generator that writes files in a `/proc` filesystem assumed to be mounted somewhere other than `/proc` so as to avoid interfering with the target container's `/proc` filesystem. This generator is intended for eventual use in `process-agent` experiments as an alternative to the existing `process_tree` generator.

### Motivation

Generate load to test `process-agent` without having to spawn processes. The existing `process_tree` generator tends to incur enough retries under some circumstances that `process-agent` experiments using that generator will time out. This behavior is undesirable because it increases the time needed to run experiments and also increases costs. Our working theory is that a `procfs` generator will be better behaved and that `process-agent` experiments using it will require fewer retries, reducing both experiment cost and time to experiment results.

### Related issues

REF SMPTNG-88.

### Additional Notes

The `procfs` generator itself is incomplete. As currently constituted, this pull request includes enough code to serialize the contents of all of the files above to strings, with the exception of `/proc/[pid]/cmdline` and `/proc/[pid]/comm`, both of which would be straightforward to implement. This pull request does *not* implement the `lading_payload::Serialize` for any `procfs` types because the typical Lading payload paradigm assumes a payload can be streamed by a generator to a blackhole as a stream of bytes that has no awareness of payload contents. A `procfs` generator needs to write certain bytes to particular locations, which would require piercing the "stream of bytes" (`lading_payload::Serialize`) abstraction. The `process_tree` generator similarly requires piercing the "stream of bytes" abstraction, which is accomplished via a "`lading`-in-`lading`" approach. This pull request defers implementing a full `procfs` generator to a later pull request because the current pull request is already pretty long. Instead, this pull request adds a stub `procfs` generator to be implemented later.

### CODE REVIEW

**Comment by blt** [comment]
@goxberry I'd argue this is related to SMPTNG-79.

**Comment by blt** [comment]
Oh sorry, I meant https://datadoghq.atlassian.net/browse/SMPTNG-88?atlOrigin=eyJpIjoiYmRiYTY0ZjEwNTdkNGIwYmE5ODVmNDg1ZDAzM2Y1ODQiLCJwIjoiaiJ9. Now assigned to you.

**Review by blt — APPROVED** [review]
There's some check dings here that need resolved but I'd say any of the comments here could be profitably pushed to follow-up PRs.

File: lading/src/generator/procfs.rs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    /// the directory tree rooted at `self.root`. Any error from
+    /// `std::io::Error` is possible.
+    pub async fn spin(mut self) -> Result<(), Error> {
+        loop {
+            tokio::select! {
+                _ = self.shutdown.recv() => {
+                    tracing::info!("shutdown signal received");
+                    break
+                },
+            }
```
[the remark this answers is not in the document]
Comment (goxberry) [reply lading/src/generator/procfs.rs]: Made this change as part of `clippy` lint cleanup in https://github.com/DataDog/lading/pull/740/commits/d68f1716b3afa9dfd8cfab87d29f63517195848f.

**Pull request merged**